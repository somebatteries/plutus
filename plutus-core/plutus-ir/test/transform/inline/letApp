(let
    (nonrec)
    (termbind (strict) (vardecl appNum (con integer)) (con integer 4))
    (termbind
      (strict)
      (vardecl addInteger (fun (con integer) (fun (con integer) (con integer))))
      (builtin addInteger)
    )
    (termbind
      (strict)
      (vardecl addInteger (fun (con integer) (fun (con integer) (con integer))))
      (lam x (con integer) (lam y (con integer) [ [ addInteger x ] y ]))
    )
    (termbind
      (strict)
      (vardecl funApp (fun (con integer) (con integer)))
      (lam
        eta
        (con integer)
        [
          [ AddInteger appNum ]
          eta
        ]
      )
    )
    [ funApp (con integer 5) ]
)