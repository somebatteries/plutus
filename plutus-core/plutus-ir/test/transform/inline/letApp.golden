[
  (lam eta (con integer) [ [ AddInteger (con integer 4) ] eta ]) (con integer 5)
]